File: Day218/palindromeString.go
package main 

import (
	"fmt"
)

func palindrome(str string) bool {
	low := 0
	high := len(str) - 1
	for low <= high {
		if str[low] != str[high] {
			return false
		}
		low++
		high--
	}
	return true;
}

func main() {
	var str string
	fmt.Scanln(&str)
	fmt.Println(palindrome(str))
}

File: Day218/palindromeString.java
package Day218;

public class palindromeString {
    public static boolean palindrome(String str) {
        int low = 0;
        int high = str.length()-1;
        while (low <= high) {
            if (str.charAt(low) != str.charAt(high)) {
                return false;
            }
            low++;
            high--;
        }
        return true;
    }
    public static void main(String[] args) {
        String str = "abcdcba";
        System.out.println(palindrome(str));    
    }
}